packages/web3torrent/src/library/web3torrent-lib.ts

```diff
@@ -172,6 +147,37 @@ export default class WebTorrentPaidStreamingClient extends WebTorrent {
     }
   }
 
+  async checkTorrentInTracker(infoHash: string) {
+    const trackerClient: Client = new Client({
```

Comment:
Why is a new client needed? Why can't we use the existing `WebTorrent` client?

Reply:
Have we tried using `noPeers` event: `torrent.on('noPeers', function (announceType) {}))` described in the [docs](https://webtorrent.io/docs)?

Reply:
> Why is a new client needed? Why can't we use the existing WebTorrent client?

I tried to, but it didn't work. I can try later again, but the issue that I had is that the test results never found peers if I did that. That's why I even make a randomized peerId, so it behaves like a different tracker client than the one on the webtorrent client.

> Have we tried using noPeers event: torrent.on('noPeers', function (announceType) {})) described in the docs?

Noup, I'll take a look

Reply:
No worries, I was mostly just wondering.

Reply:
decided to test and touch more stuff tomorrow, so we can merge this one, and I don't block your refactor.